src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramParser.java

```diff
@@ -46,6 +46,12 @@
     static final ParseField OFFSET = new ParseField("offset");
     static final ParseField PRE_OFFSET = new ParseField("", "pre_offset");
     static final ParseField POST_OFFSET = new ParseField("", "post_offset");
+    static final ParseField PRE_ZONE = new ParseField("", "pre_zone");
+    static final ParseField POST_ZONE = new ParseField("", "post_zone");
```

Comment:
I could be wrong but it looks like the argument to withAllDeprecated indicates the new setting which has replaced the deprecated options and is only used in the error message to point the user to the new option. so for `pre_zone` we would use `new ParseField("pre_zone").withAllDeprecated("time_zone")`. It might be worth checking with @s1monw that this is indeed how it should work